app/services/course/assessment/question/programming/python/python_package_service.rb

```diff
@@ -198,6 +174,28 @@ def generate_zip_file(data_files_to_keep)
     tmp
   end
 
+  def zip_test_files(test_type, zip)
```

Comment:
This is probably as short and concise as you can have it be, suggest you disable rubocop for this case - will have to disable 2. 

The only other thing I can think of simplifying is 
```ruby
  zip.print <<-PYTHON
def test_#{test_type}_#{format('%02i', index)}(self):
    self.meta['expression'] = #{test[:expression].inspect}
    self.meta['expected'] = #{expected}
    #{hint}
     _out = #{test[:expression]}
    self.meta['output'] = _out
    self.assertEqual(_out, #{test[:expected]})
  PYTHON
```